elrond-wasm-node/src/api/send_api_node_managed.rs

```diff
@@ -0,0 +1,500 @@
+        arg_buffer: &ManagedArgBuffer<Self::ProxyTypeManager>,
+        range_closure: F,
+    ) -> ManagedVec<Self::ProxyTypeManager, ManagedBuffer<Self::ProxyTypeManager>>
+    where
+        F: FnOnce(usize, usize) -> (usize, usize),
+    {
+        unsafe {
+            let num_return_data_before = getNumReturnData() as usize;
+            let result_handle = mBufferNew();
+
+            let _ = managedExecuteOnDestContext(
+                gas as i64,
+                to.get_raw_handle(),
+                amount.get_raw_handle(),
+                endpoint_name.get_raw_handle(),
+                arg_buffer.get_raw_handle(),
+                result_handle,
+            );
+
+            let result = ManagedVec::from_raw_handle(self.clone(), result_handle);
+
+            let num_return_data_after = num_return_data_before + result.len();
+            let (result_start_index, result_end_index) = range_closure(
+                num_return_data_before as usize,
+                num_return_data_after as usize,
+            );
+            result
+                .slice(
+                    result_start_index - num_return_data_before,
+                    result_end_index - num_return_data_before,
+                )
+                .unwrap_or_else(|| error_hook::signal_error(b"sync call range bad slicing"))
+        }
+    }
+
+    fn execute_on_dest_context_by_caller_raw(
+        &self,
+        gas: u64,
+        to: &ManagedAddress<Self::ProxyTypeManager>,
+        amount: &BigUint<Self::ProxyTypeManager>,
+        endpoint_name: &ManagedBuffer<Self::ProxyTypeManager>,
+        arg_buffer: &ManagedArgBuffer<Self::ProxyTypeManager>,
+    ) -> ManagedVec<Self::ProxyTypeManager, ManagedBuffer<Self::ProxyTypeManager>> {
+        unsafe {
+            let result_handle = mBufferNew();
+
+            let _ = managedExecuteOnDestContextByCaller(
+                gas as i64,
+                to.get_raw_handle(),
+                amount.get_raw_handle(),
+                endpoint_name.get_raw_handle(),
+                arg_buffer.get_raw_handle(),
+                result_handle,
+            );
+
+            ManagedVec::from_raw_handle(self.clone(), result_handle)
+        }
+    }
+
+    fn execute_on_same_context_raw(
+        &self,
+        gas: u64,
+        to: &ManagedAddress<Self::ProxyTypeManager>,
+        amount: &BigUint<Self::ProxyTypeManager>,
+        endpoint_name: &ManagedBuffer<Self::ProxyTypeManager>,
+        arg_buffer: &ManagedArgBuffer<Self::ProxyTypeManager>,
+    ) {
+        unsafe {
+            let result_handle = mBufferNew();
+
+            let _ = managedExecuteOnSameContext(
+                gas as i64,
+                to.get_raw_handle(),
+                amount.get_raw_handle(),
+                endpoint_name.get_raw_handle(),
+                arg_buffer.get_raw_handle(),
+                result_handle,
+            );
+
+            // TODO: return result?
```

Comment:
Yes, this should definitely return the results, just like executeOnDestContext.

Reply:
Next PR, I also need to ad tests